mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAModSeqProvider.java

```diff
@@ -59,7 +62,7 @@ public long highestModSeq(MailboxSession session, Mailbox mailbox) throws Mailbo
             }
             throw new MailboxException("Unable to get highest mod-sequence for mailbox " + mailbox, e);
         } finally {
-            if (manager != null) {
+            if (manager != null && !manager.getTransaction().isActive()) {
```

Comment:
could you explain this change? why do you deny to close manager if the transaction is not active?

Reply:
when an unchecked exception be thrown and in finally block, if the transaction state now is still active, an InvalidStateException with message "The operation cannot be performed while a Transaction is active" will be thrown.